app/routes.py

```diff
@@ -1 +1,217 @@
-from flask import Blueprint
\ No newline at end of file
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+from flask import Blueprint, jsonify, request, make_response, abort
+from datetime import datetime
+import requests
+import os
+from dotenv import load_dotenv
+
+load_dotenv()
+
+task_bp = Blueprint("task_bp", __name__, url_prefix="/tasks")
+goal_bp = Blueprint("goal_bp", __name__, url_prefix="/goals")
+
+def validate_model(cls, model_id):
+    try:
+        model_id = int(model_id)
+    except:
+        abort(make_response({"message":f"{cls.__name__} {model_id} invalid"}, 400))
+
+    model = cls.query.get(model_id)
+
+    if not model:
+        abort(make_response({"message":f"{cls.__name__} {model_id} not found"}, 404))
+
+    return model
+
+
+@task_bp.route("", methods=["POST"])
+def handle_tasks():
+    request_body = request.get_json()
+    try:
+        new_task = Task(title=request_body["title"],
+                    description=request_body["description"])
+    except:
+        abort(make_response({"details": "Invalid data"}, 400))
+
+    db.session.add(new_task)
+    db.session.commit()
+
+    return make_response({"task":{
+            "id": new_task.task_id,
+            "title": new_task.title,
+            "description": new_task.description,
+            "is_complete": False
+        }}, 201)
+
+@task_bp.route("", methods=["GET"])
+def read_all_tasks():
+
+    sort_query = request.args.get("sort")
+
+    if sort_query == "asc":
+        task = Task.query.order_by(Task.title.asc())
+    elif sort_query == "desc":
+        task = Task.query.order_by(Task.title.desc())
+    else:
+        task = Task.query.all()
+
+    task_response = []
+    for task in task:
+        task_response.append(task.to_dict())
+        
+    return jsonify(task_response)
+
+@task_bp.route("/<task_id>", methods=["GET"])
+def read_one_task(task_id):
+    task = validate_model(Task, task_id)
+    return {"task": task.to_dict()}
+
+@task_bp.route("/<task_id>", methods=["PUT"])
+def update_task(task_id):
+    task = validate_model(Task, task_id)
+
+    request_body = request.get_json()
+
+    task.title = request_body["title"]
+    task.description = request_body["description"]
```

Comment:
We should have error handling here (like you did with your POST request method). We can't rely on the client always sending a proper request body.